client.py

```diff
@@ -146,25 +149,71 @@ async def handle_add_game_request(message, game_name):
         reply = 'This game doesn\'t exist on steam, try gamepass.'
     await message.reply (reply)
 
+async def handle_delete_game_request_from_match(message, index_string):
+    index = int(index_string) - 1
+    game_matches = client_state.get_game_matches()
+    if len(game_matches) == 0:
+        print('game matches is empty')
+        return
+    game_tupple = game_matches[index]
+    deleted_game_result = delete_game(game_tupple)
+    print('Deleted game result is:', deleted_game_result)
+    if deleted_game_result == True:
+        reply = f'**{game_tupple[1]}** was successfully deleted from your wishlist.'
+    await message.reply(reply)
+
 # Function to handle client side of a delete game request, builds reply with list_games_for_reply function
-async def handle_delete_game_request(message, entries):
+async def handle_delete_game_request(message, game_name):
     reply = ''
-    if len(entries) == 1:
-        entry = entries[0]
-        status = delete_game((entry["appid"], entry["name"]))
+    matches = get_entries_from_wishlist(game_name)
+    if len(matches) == 1:
+        # Working with a list with a single tupple in it from matches
+        tupple = (matches[0][0], matches[0][1])
+        status = delete_game(tupple)

```

Comment:
Won't this always delete the game, even if the entered value doesn't match exactly? So the IF-ELSE below will always be true unless some other failure occurs?

Reply:
Yup this deletes the game if only 1 exists with "star wars" in it, sort of assuming that was the correct one. Is this dangerous? In the add we have the "is this the game you meant prompt but wasn't sure how to implement that for delete (we spoke about having an if for which side the argument came from add or delete)

Reply:
Yeah we could add state to the ClientStateManager to keep track of which operation we are storing the list/handling stuff for